server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -678,19 +679,27 @@ private Settings getOverrideSettingsInternal() {
 
                         // We will use whatever replication strategy provided by user or from snapshot metadata unless
                         // cluster is remote store enabled or user have restricted a specific replication type in the
-                        // cluster.
+                        // cluster. If cluster is undergoing remote store migration, replication strategy is strictly SEGMENT type
                         if (RemoteStoreNodeAttribute.isRemoteStoreAttributePresent(clusterService.getSettings()) == true
-                            || clusterSettings.get(IndicesService.CLUSTER_INDEX_RESTRICT_REPLICATION_TYPE_SETTING) == true) {
+                            || clusterSettings.get(IndicesService.CLUSTER_INDEX_RESTRICT_REPLICATION_TYPE_SETTING) == true
+                            || RemoteStoreNodeService.isMigratingToRemoteStore(clusterSettings) == true) {
```

Comment:
nit : lets remove `==true` from all checks. We can  do it in a follow up PR as well .